From c9a830e0e4abb73f22b8c4525bc26685cb59076a Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Fri, 4 Apr 2014 18:23:16 +0800
Subject: [PATCH 3882/5965] PD #83628: driver update for PMU2/PMU3

1. set dcdc3 duty cycle limit according CHG_GAT_BAT_LV bit;
---
 drivers/amlogic/power/aml_pmu/aml1216-sply.c | 4 ++++
 drivers/amlogic/power/aml_pmu/aml1218-sply.c | 6 +++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/power/aml_pmu/aml1216-sply.c b/drivers/amlogic/power/aml_pmu/aml1216-sply.c
index 8015643855e2..a1bacd2d1227 100755
--- a/drivers/amlogic/power/aml_pmu/aml1216-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1216-sply.c
@@ -1193,6 +1193,10 @@ static int aml1216_update_state(struct aml_charger *charger)
     charger->usb_valid  = (val & 0x08) ? 1 : 0; 
     charger->ext_valid  = charger->dcin_valid | (charger->usb_valid << 1); 
     charger->fault      = val;
+    /*
+     * limit duty cycle of DC3 according CHG_GAT_BAT_LV bit
+     */
+    aml1216_set_bits(0x004f, (val & 0x01) << 3, 0x08);
 
     charger->vbat = aml1216_get_battery_voltage();
     charger->ocv  = aml1216_cal_ocv(charger->ibat, charger->vbat, charger->charge_status);
diff --git a/drivers/amlogic/power/aml_pmu/aml1218-sply.c b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
index 4886ae7e1d81..2e71497313c3 100755
--- a/drivers/amlogic/power/aml_pmu/aml1218-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
@@ -1177,7 +1177,7 @@ static int aml1218_update_state(struct aml_charger *charger)
     uint32_t chg_status;
 
     aml1218_read(0x00E0, &val);
-    aml1218_reads(0x00df, &chg_status, 4);
+    aml1218_reads(0x00de, &chg_status, 4);
 
     charger->ibat = aml1218_get_battery_current();
     if (val & 0x18) {
@@ -1194,6 +1194,10 @@ static int aml1218_update_state(struct aml_charger *charger)
     charger->usb_valid  = (val & 0x08) ? 1 : 0; 
     charger->ext_valid  = charger->dcin_valid | (charger->usb_valid << 1); 
     charger->fault      = chg_status;
+    /*
+     * limit duty cycle of DC3 according CHG_GAT_BAT_LV bit
+     */
+    aml1218_set_bits(0x004f, (chg_status & 0x02000000) >> 22, 0x08);
 
     charger->vbat = aml1218_get_battery_voltage();
     charger->ocv  = aml1218_cal_ocv(charger->ibat, charger->vbat, charger->charge_status);
-- 
2.19.0

